### ISSUE TITLE
I have a react component that updates frequently and it seems to put after.js into an infinite loop

### ISSUE THREAD

**Issue by scottdickerson** [dequelabs/react-axe#19]
I have a react component that frequently updates when you mouseover a date region, it's similar to the contributions view of github:

<img width="754" alt="screen shot 2017-08-17 at 5 21 52 pm" src="https://user-images.githubusercontent.com/6663002/29419861-979a193c-8370-11e7-9dcb-2ba6fd971000.png">

For some reason when react-axe is active and I mouseover it and move my mouse it seems to get into an infinite loop in the after.js on the originalFn.apply line.

```var restoreFunctions = [];

var after = function after(host, name, cb) {
	var originalFn = host[name];
	var restoreFn = undefined;

	if (originalFn) {
		host[name] = function () {
			originalFn.apply(this, arguments);
			cb(host);
		};
		restoreFn = function () {
			return host[name] = originalFn;
		};
	} else {
		host[name] = function () {
			cb(host);
		};
		restoreFn = function () {
			return delete host[name];
		};
	}

	restoreFunctions.push(restoreFn);
};

after.restorePatchedMethods = function () {
	restoreFunctions.forEach(function (restoreFn) {
		return restoreFn();
	});
	restoreFunctions = [];
};

module.exports = after;
```
```
Uncaught RangeError: Maximum call stack size exceeded
    at DataBar.host.(anonymous function) (http://localhost:3000/js/vendor_bundled_22229635cd1f11a8b467.js?826970b10a47f95bba4e:110279:25)
    at DataBar.h
```

Can I somehow disable react-axe just for this component with the configuration options?  Failing that, any other ideas?

**Comment by swiftfoot** [comment]
I'm react-axe 2.1.7

**Comment by dylanb** [comment]
any chance you could reproduce this in a codepen?

**Comment by swiftfoot** [comment]
I'll try to, do you have an external javascript version of react-axe that I can link to from a CDN.  I believe it should happen within this JSFiddle once you add react-axe wrapping:   

http://jsfiddle.net/x1yoboc7/963/

I was seeing the infinite loop when mousing over one of the points on the line graph.

**Comment by HugoGiraudel** [comment]
We have something similar, however I couldn’t reproduce it on a fresh `create-react-app`. :(

<img width="566" alt="screen shot 2018-04-24 at 11 51 56 am" src="https://user-images.githubusercontent.com/1889710/39179959-f6cd9712-47b5-11e8-8e23-bec7f1e1ef2a.png">

**Comment by KirillKayumov** [comment]
We're on `react-axe` 3.0.1 and we experience the same problem as well. Every time the UI is re-rendered our application is becoming slower and slower, and at some point, it starts producing `Maximum call stack size exceeded` error.

**Comment by HugoGiraudel** [comment]
In our side, the problem was related to react-helmet doing permanent rerenders. Are you using it as well?

**Comment by KirillKayumov** [comment]
@HugoGiraudel no, we don't. But the problem appears when UI is complex and many components are re-rendered.

**Comment by alexzherdev** [comment]
I believe this was fixed in #44.

**Comment by straker** [comment]
Closing as no more discussion has occurred. Assuming #44 fixed the issue as suggested.
